nutils/evaluable.py

```diff
@@ -2291,6 +2420,20 @@ def _sign(self):
     if self.dofmap.isconstant and _isunique(self.dofmap.eval()):
       return Inflate(Sign(self.func), self.dofmap, self.length)
 
+  @property
+  def _assparse(self):
+    chunks = []
+    flat_dofmap = _ravel_all(self.dofmap)
+    keep_dim = self.func.ndim - self.dofmap.ndim
+    strides = (1, *itertools.accumulate(self.dofmap.shape[:0:-1], operator.mul))[::-1]
+    for *indices, values in self.func._assparse:
+      if self.dofmap.ndim:
+        inflate_indices = Take(flat_dofmap, functools.reduce(operator.add, map(operator.mul, indices[keep_dim:], strides)))
```

Comment:
That's because the `indices[keep_dim:]` don't necessarily have a tensor structure. With a tensor structure we could write the `inflate_indices` as

```python
inflate_indices = self.dofmap
for iaxis, indices in enumerate(indices[keep_dim:]):
  indices = detensor(indices, keep_dim+iaxis)
  inflate_indices = _take(inflate_indices, indices, iaxis)
```

where `detensor` uninserts all axes except the given axis.

Reply:
Check, sorry, failed to see that we're only flattening the inflated axis.